### PR TITLE
jetbrains: fix CVE-2024-37051

### PR DESCRIPTION
**Pull request by kashw2**
## Description of changes

ran `update_bin.py`

Fixes https://www.cve.org/CVERecord?id=CVE-2024-37051
https://blog.jetbrains.com/security/2024/06/updates-for-security-issue-affecting-intellij-based-ides-2023-1-and-github-plugin/

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/editors/jetbrains/bin/versions.json [hunk 1]
```diff
@@ -166,91 +166,91 @@
...
     },
     "mps": {
       "update-channel": "MPS RELEASE",
       "url-template": "https://download.jetbrains.com/mps/{versionMajorMinor}/MPS-{version}.tar.gz",
-      "version": "2023.3",
+      "version": "2023.3.1",
       "sha256": "7f40c6a9b29e17e29fd2acb2e0d1d31d3353e28fce31a479dd81dd1b66c1fa8c",
-      "url": "https://download.jetbrains.com/mps/2023.3/MPS-2023.3.tar.gz",
-      "build_number": "233.13135.979"
+      "url": "https://download.jetbrains.com/mps/2023.3.1/MPS-2023.3.1.tar.gz",
```
Comment (zlepper) [inline pkgs/applications/editors/jetbrains/bin/versions.json:195]: Mps fails to download for me. The correct url is `https://download.jetbrains.com/mps/2023.3/MPS-2023.3.1.tar.gz`

For some reason it's not available under the 2023.3.1 folder :/

**Comment by zlepper** [comment]
Result of `nixpkgs-review pr 319290` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>jetbrains.mps</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.datagrip</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

File: pkgs/applications/editors/jetbrains/bin/versions.json (hunk 1 above)
Comment (kashw2) [reply pkgs/applications/editors/jetbrains/bin/versions.json:195]: Odd, should be fixed in https://github.com/NixOS/nixpkgs/pull/319290/commits/40dfc806d6a185257e2bb67b6704c0e15001a763

Comment (leona-ya) [reply pkgs/applications/editors/jetbrains/bin/versions.json:195]: Now it fails with 

```
error: hash mismatch in fixed-output derivation '/nix/store/9zzllgyva59jrdkdfj01xil8ag4i1x6l-MPS-2023.3.1.tar.gz.drv':
         specified: sha256-f0DGqbKeF+Kf0qyy4NHTHTNT4o/OMaR53YHdG2bB+ow=
            got:    sha256-UqQP4F3Y1To/aPqnpuQLc/cvgNtZdtjr14GVVJOmX+E=

Comment (leona-ya) [reply pkgs/applications/editors/jetbrains/bin/versions.json:195]: So probably the hash should be `52a40fe05dd8d53a3f68faa7a6e40b73f72f80db5976d8ebd781955493a65fe1`, but please test again.

Comment (kashw2) [reply pkgs/applications/editors/jetbrains/bin/versions.json:195]: Updated the hash with what's found at `https://download.jetbrains.com/mps/2023.3/MPS-2023.3.1.tar.gz.sha256`, built with `nix build .#jetbrains.mps` and it succeeds.

**Review by leona-ya — APPROVED** [review]
Thank you very much!

**Pull request merged**